src/app/screens/connectors/ConnectAlby/index.tsx

```diff
@@ -0,0 +1,82 @@
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useNavigate } from "react-router-dom";
+import { toast } from "react-toastify";
+import Button from "~/app/components/Button";
+import msg from "~/common/lib/msg";
+import { WebLNNode } from "~/extension/background-script/connectors/connector.interface";
+
+type Props = {
+  fromWelcome?: boolean;
+};
+
+export default function ConnectAlby({ fromWelcome }: Props) {
+  const [loading, setLoading] = useState(false);
+  const navigate = useNavigate();
+  const { t: tCommon } = useTranslation("common");
+  async function connectAlby() {
+    setLoading(true);
+    const name = "Alby";
+    const initialAccount = {
+      name,
+      config: {},
+      connector: "alby",
+    };
+
+    try {
+      const validation = await msg.request("validateAccount", initialAccount);
+      if (validation.valid) {
+        if (!validation.oAuthToken) {
+          throw new Error("No oAuthToken returned");
+        }
+
+        const account = {
+          ...initialAccount,
+          name: (validation.info as { data: WebLNNode }).data.alias,
+          config: {
+            ...initialAccount.config,
+            oAuthToken: validation.oAuthToken,
+          },
+        };
+
+        const addResult = await msg.request("addAccount", account);
+        if (addResult.accountId) {
+          await msg.request("selectAccount", {
+            id: addResult.accountId,
+          });
+          if (fromWelcome) {
```

Comment:
if the user already has an account and tries to add an alby account it will go to discover as expected. but the discover page is rendered under the name of the same existing account and not the new account. i had to refresh to make existing account visible under extension

Reply:
i think this is happening because the way we persisting the account in the connector and setting the state of the account.

```
 if (this.account.id) {
        const accounts = state.getState().accounts;
        const password = (await state.getState().password()) as string;

        const configData = decryptData(
          accounts[this.account.id].config,
          password
        );
        configData.oAuthToken = this.config.oAuthToken;
        accounts[this.account.id].config = encryptData(configData, password);
        state.setState({ accounts });
        // make sure we immediately persist the updated accounts
        await state.getState().saveToStorage();
      }
```

also this part is not executed when we add account. during that time this.account.id will be undefined. is this valid behaviour @rolznz ?

Reply:
@pavanjoshi914 I think it is because it no longer uses the TestConnection code which seems to update the account list and set the selected account to the newly created account. I still don't think it makes sense to test the connection for the Alby connector because we already successfully logged in, but we need the functionality to select and update the accounts. Could you have a look at this?

Reply:
checking! but how we are persisting tokens for the first time we add account as the above code is not executed

Reply:
@pavanjoshi914 the account is added, but the accounts list and selected account is not updated in the frontend